packages/runtime-common/realm-index-updater.ts

```diff
@@ -169,27 +173,44 @@ export class RealmIndexUpdater {
   }
 
   private async getRealmUsername(): Promise<string> {
-    // TODO for now we are just using the URL pattern and hard coding test
-    // URLs to figure out the realm username. As part of the ticket to create
-    // dynamic realms this should be updated to look up the realm owner
-    // permission
+    let permissions = await fetchUserPermissions(
+      this.#dbAdapter,
+      this.realmURL,
+    );
+    let owners = Object.entries(permissions)
+      .filter(([_, permissions]) => permissions.includes('realm-owner'))
+      .map(([userId]) => userId);
+    let realmUserId =
+      owners.length === 1
+        ? owners[0]
+        : owners.find((userId) => userId.startsWith('@realm/'));
+    if (realmUserId) {
+      return getMatrixUsername(realmUserId);
+    }
+
+    // hard coded test URLs
```

Comment:
What is missing about how we set up test realms that we need this mapping? A realm owner?

Reply:
Yeah. Most tests do not set this. Altho, it only really matters for realms that are. It public, otherwise for world readable realms (which are lost of our tests), auth is entirely bypassed during indexing.